### PR TITLE
Add benchmarks for logic module

### PR DESCRIPTION
**Pull request by debugger22**
I haven't tested the code because it gives ProcessError(timed out) on my system.

Original benchmarks are here https://github.com/sympy/sympy/tree/master/sympy/logic/benchmarks

@moorepants @bjodah

### CODE REVIEW

File: benchmarks/logic.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class LogicSuite:
+    """
+    Benchmark suite for sympy.logic
+    """
+    def setup(self):
+        self.INPUT = [5 * i for i in range(2, 16)]
+        self.ALGORITHMS = ['dpll', 'dpll2']
+
```
Comment (bjodah) [inline benchmarks/logic.py]: Maybe you want to see differences between different algorithms? If so you can introduce `time_dpll(self)` and `time_dpll2(self)`

**Comment by bjodah** [comment]
I get the following error message when running `asv run` on your branch:

```
$ ~/.local/bin/asv run
· Fetching recent changes.
· Creating environments
· No __init__.py file in 'benchmarks/logic-inputs'
```
Same error in asv 0.1.1 and asv master branch.

I see you have quite a few cnf files, do you vision them ever being changed? if not you could put them in a .tar.gz or zip file and have them extracted into a [temporary directory](https://docs.python.org/2/library/tempfile.html#tempfile.mkdtemp) upon `setup(self)`.

So you employ a SAT solver in logic module, what about optionally employing `pycosat` much like `fastcache` is an optional dependency?

File: benchmarks/logic.py (hunk 1 above)
Comment (debugger22) [inline benchmarks/logic.py]: That would be better.

**Comment by debugger22** [comment]
> I see you have quite a few cnf files, do you vision them ever being changed? if not you could put them in a .tar.gz or zip file and have them extracted into a temporary directory upon setup(self).

I'll do that.

> So you employ a SAT solver in logic module, what about optionally employing pycosat much like fastcache is an optional dependency?

I created an issue https://github.com/sympy/sympy/issues/9728 to discuss.

**Comment by debugger22** [comment]
@bjodah Could you please test if this works? I added a mock `__init__.py` in that directory.

File: benchmarks/logic.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    """
+    Benchmark suite for sympy.logic
+    """
+    def setup(self):
+        self.INPUT = [5*i for i in range(2, 16)]
+
+    def time_dpll(self):
+        for test in self.INPUT:
+            file_name = os.path.join(input_path, 'logic-inputs', '%s.cnf' % test)
+            theory = load_file(file_name)
```
Comment (bjodah) [inline benchmarks/logic.py]: Maybe all files should be loaded in `setup(self)`? (in the event that loading would be slow)
You could also benchmark the loading a "time_load" which loads a single file.

**Comment by bjodah** [comment]
Great, `asv run` now works as expected.

File: benchmarks/logic.py (hunk 2 above)
Comment (debugger22) [inline benchmarks/logic.py]: +1

**Comment by asmeurer** [comment]
Great. It's actually more important that we be benchmarking functions like `to_cnf`, which are often the real bottleneck.

**Comment by debugger22** [comment]
@bjodah Perhaps this is ready to go.

**Comment by bjodah** [comment]
Yes, look good. Thanks!

**Comment by debugger22** [comment]
Thanks for the review!

**Pull request merged**